tempest/api/workloadmgr/multiattach_volumes/test_multiattach_volumes.py

```diff
@@ -0,0 +1,703 @@
+            self.vm_id_1 = self.create_vm(key_pair=self.kp)
+            self.vm_id_2 = self.create_vm(key_pair=self.kp)
+            mount_point = "mount_data_b"
+
+            # find volume_type = multiattach. So that existing multiattach volume type can be used.
+            # Get the volume_type_id
+            vol_type_id = -1
+            for vol in CONF.volume.volume_types:
+                if (vol.lower().find("multiattach") != -1):
+                    vol_type_id = CONF.volume.volume_types[vol]
+                    vol_type_name = vol
+
+            if (vol_type_id == -1):
+                raise Exception("No multiattach volume found to create multiattach volume. Test cannot be continued")
+
+            # Now create volume with derived volume type id...
+            self.volume_id = self.create_volume(
+                volume_type_id=vol_type_id, size=10)
+
+            LOG.debug("Volume ID: " + str(self.volume_id))
+
+            self.volumes = []
+            self.volumes.append(self.volume_id)
+            # Attach volume to vm...
+            self.attach_volume(self.volume_id, self.vm_id_1)
+            self.attach_volume(self.volume_id, self.vm_id_2,attach_cleanup=False)
+            LOG.debug("Multiattach Volume attached to vm: " + str(self.vm_id_1) + " and " + str(self.vm_id_2))
+            self.disk_names = ["vdb"]
+
+            vol_vm_1 = self.get_attached_volumes(self.vm_id_1)
+            vol_vm_2 = self.get_attached_volumes(self.vm_id_2)
+            LOG.debug("Voulme o VM 1: " + str(vol_vm_1) + " on VM 2:" + str(vol_vm_2))
+            if vol_vm_1 == vol_vm_2:
+                reporting.add_test_step("Attached Multiattach volume to both Instances", tvaultconf.PASS)
+                reporting.set_test_script_status(tvaultconf.PASS)
+            else:
+                reporting.add_test_step("Attached Multiattach volume to both Instances", tvaultconf.FAIL)
+                reporting.set_test_script_status(tvaultconf.FAIL)
+                raise Exception("Multiattach volume failed to attach existing instance")
+            fip = self.get_floating_ips()
+            LOG.debug("\nAvailable floating ips are {}: \n".format(fip))
+            # if len(fip) < 8:
+            if len(fip) < 6:
+                raise Exception("Floating ips unavailable")
+            self.set_floating_ip(fip[0], self.vm_id_1)
+            self.set_floating_ip(fip[1], self.vm_id_2)
+
+            # self.frm_id = self.create_vm(
+            #     vm_name="file_recovery_manager",
+            #     flavor_id=CONF.compute.flavor_ref_alt,
+            #     user_data=tvaultconf.user_frm_data,
+            #     key_pair=self.kp,
+            #     image_id=list(CONF.compute.fvm_image_ref.values())[0])
+            # self._set_frm_user()
+            # LOG.debug("FRM Instance ID: " + str(self.frm_id))
+            # self.set_floating_ip(fip[2], self.frm_id)
+
+            ssh = self.SshRemoteMachineConnectionWithRSAKey(fip[0])
```

Comment:
Connecting to remote machine and executing multiple commands + adding files is repetitive code. Please create common function and call at required places.

Reply:
added common function